### PR TITLE
feature/with tuples

### PR DESCRIPTION
**Pull request by rhayes777**
Some improvements but I don't think it resolves the problem raised in #450

### CODE REVIEW

**Comment by Jammy2211** [comment]
```
samples = samples.with_paths(
    ["galaxies.lens..mass.einstein_radius", "galaxies.source.bulge.sersic_index"]
)
```

Does the PR aim to support an API like this? (I thought I saw code for this).

I cant get it work using the above syntax. Is this syntax slightly wrong? And if so can we make it support this as I think this is the most concise syntax and is most in line with model composition.

File: autofit/mapper/prior_model/abstract.py [hunk 1]
```diff
@@ -248,7 +251,7 @@ def _with_paths(
 
     def with_paths(
             self,
-            paths: List[Tuple[str]]
+            paths: List[Tuple[str, ...]]
```
Comment (Jammy2211) [inline autofit/mapper/prior_model/abstract.py:256]: I am getting lint errors when there is more than 1 string in my tuple specifying paths.

File: autofit/mapper/prior_model/abstract.py [hunk 2]
```diff
@@ -315,7 +321,7 @@ def _without_paths(
 
     def without_paths(
             self,
-            paths: List[Tuple[str]]
+            paths: List[Tuple[str, ...]]
```
Comment (Jammy2211) [inline autofit/mapper/prior_model/abstract.py:327]: Same here.

File: autofit/mapper/prior_model/abstract.py (hunk 1 above)
Comment (rhayes777) [reply autofit/mapper/prior_model/abstract.py:256]: Yeah the ... should fix that

**Review by Jammy2211 — APPROVED** [review]

**Pull request merged**